File: ios/BitmovinCastManagerModule.swift
import BitmovinPlayer
import ExpoModulesCore

public class BitmovinCastManagerModule: Module {
    public func definition() -> ModuleDefinition {
        Name("BitmovinCastManagerModule")

        AsyncFunction("isInitialized") { () -> Bool in
            #if os(iOS)
            return BitmovinCastManager.isInitialized()
            #else
            return false
            #endif
        }.runOnQueue(.main)

        AsyncFunction("sendMessage") { (message: String, messageNamespace: String?) in
            #if os(iOS)
            BitmovinCastManager.sharedInstance().sendMessage(message, withNamespace: messageNamespace)
            #endif
        }.runOnQueue(.main)

        AsyncFunction("updateContext") { () in
            // iOS/tvOS doesn't need updateContext like Android does
            // This is a no-op for iOS compatibility
        }.runOnQueue(.main)
    }
}

File: android/src/main/java/com/bitmovin/player/reactnative/BitmovinCastManagerModule.kt
package com.bitmovin.player.reactnative

import android.util.Log
import com.bitmovin.player.casting.BitmovinCastManager
import com.bitmovin.player.reactnative.extensions.getString
import expo.modules.kotlin.functions.Queues
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class BitmovinCastManagerModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("BitmovinCastManagerModule")

        AsyncFunction("isInitialized") {
            BitmovinCastManager.isInitialized()
        }

        AsyncFunction("sendMessage") { message: String, messageNamespace: String? ->
            BitmovinCastManager.getInstance().sendMessage(message, messageNamespace)
        }.runOnQueue(Queues.MAIN)

        AsyncFunction("updateContext") {
            appContext.currentActivity?.let { activity ->
                BitmovinCastManager.getInstance().updateContext(activity)
            }
        }.runOnQueue(Queues.MAIN)
    }
}
